src/KnowbaseItem.php

```diff
@@ -572,22 +572,66 @@ public static function getVisibilityCriteria(bool $forceall = false): array
     {
         global $CFG_GLPI;
 
+        // Build common JOIN clause
+        $criteria = [
+            'LEFT JOIN' => self::getVisibilityCriteriaCommonJoin($forceall),
+        ];
+
+        // Handle anonymous users
+        if (!Session::getLoginUserID()) {
+            if ($CFG_GLPI["use_public_faq"]) {
+                // Public FAQ is enabled; show FAQ
+                $criteria['WHERE'] = self::getVisibilityCriteriaFAQ();
+                return $criteria;
+            } else {
+                // Public FAQ is disabled; show nothing
+                $criteria['WHERE'] = [0];
+                return $criteria;
+            }
```

Comment:
for readability : limiting nest level 

```suggestion
            }
             // Public FAQ is disabled; show nothing
             $criteria['WHERE'] = [0];
             return $criteria;
```

Reply:
If/else is more readable to me, I don't think this is important.